docs/storybook/src/components/PropertyGrid.stories.tsx

```diff
@@ -46,3 +54,279 @@ export const Basic: Story = {
     height: 400,
   },
 };
+
+const rendererManager = new PropertyValueRendererManager();
+
+rendererManager.registerRenderer("customRendererStructPropertyRenderer", {
+  canRender: () => true,
+  render: (record) => {
+    const entries = Object.entries((record.value as StructValue).members);
+    return (
+      <ul>
+        {entries.map((entry) => {
+          return (
+            <li key={entry[0]}>
+              {entry[0]} = {(entry[1].value as PrimitiveValue).displayValue}
+            </li>
+          );
+        })}
+      </ul>
+    );
+  },
+});
+
+rendererManager.registerRenderer("customRendererArrayPropertyRenderer", {
+  canRender: () => true,
+  render: (record) => {
+    const items = (record.value as ArrayValue).items;
+    return (
+      <ol>
+        {items.map((item, index) => {
+          return (
+            <li key={index}>{(item.value as PrimitiveValue).displayValue}</li>
+          );
+        })}
+      </ol>
+    );
+  },
+});
+
+rendererManager.registerRenderer("defaultRendererPropertyRenderer", {
+  canRender: () => false,
+  render: () => {
+    <div>Should not render</div>;
+  },
+});
+
+rendererManager.registerRenderer(
+  "multiline",
+  new MultilineTextPropertyValueRenderer()
+);
+
+const structMembers = {
+  member1: PropertyRecord.fromString("Value 1"),
+  member2: PropertyRecord.fromString("Value 2"),
+  member3: PropertyRecord.fromString("Value 3"),
+};
+
+const arrayMembers = [
+  PropertyRecord.fromString("Value 1"),
+  PropertyRecord.fromString("Value 2"),
+  PropertyRecord.fromString("Value 3"),
+];
+
+export const StructRendering: Story = {
+  args: {
+    propertyValueRendererManager: rendererManager,
+    dataProvider: {
+      getData: async () => ({
+        label: PropertyRecord.fromString("Record 1"),
+        categories: [
+          {
+            name: "structPropertyRendering",
+            label: "Struct property rendering",
+            expand: true,
+          },
+        ],
+        records: {
+          structPropertyRendering: [
+            {
+              value: {
+                valueFormat: PropertyValueFormat.Primitive,
+                value: "string value",
+              },
+              property: {
+                name: "simpleProperty",
+                displayLabel: "Simple property",
+                typename: "string",
+              },
+              copyWithNewValue: () => ({} as unknown as PropertyRecord),
+              getChildrenRecords: () => [],
+            },
```

Comment:
You should create `PropertyRecord` class instance instead of constructing plain object. `new PropertyRecord({ ...value props... }, { ...property props... })`